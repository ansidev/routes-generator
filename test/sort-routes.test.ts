import sortRoutes from '../src/sort-routes'

test('simple', () => {
  const result = sortRoutes([
    {
      path: '/',
    },
    {
      path: '/:user'
    },
    {
      path: '/post/bar'
    },
    {
      path: '/post/:id'
    },
    {
      path: '/post/foo'
    }
  ])

  expect(result).toMatchSnapshot()
})
