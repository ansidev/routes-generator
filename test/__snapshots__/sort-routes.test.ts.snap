// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple 1`] = `
Array [
  Object {
    "index": 0,
    "route": Object {
      "path": "/",
    },
    "score": 104,
  },
  Object {
    "index": 2,
    "route": Object {
      "path": "/post/bar",
    },
    "score": 14,
  },
  Object {
    "index": 4,
    "route": Object {
      "path": "/post/foo",
    },
    "score": 14,
  },
  Object {
    "index": 3,
    "route": Object {
      "path": "/post/:id",
    },
    "score": 13,
  },
  Object {
    "index": 1,
    "route": Object {
      "path": "/:user",
    },
    "score": 6,
  },
]
`;
